services/tokenizer/basic/tokenizer.py

```diff
@@ -219,15 +219,32 @@ def _is_url_like(self, token: str) -> bool:
         if self._is_email_like(token):
             return False
 
-        return (
-            token.startswith(("http://", "https://", "www."))
-            or "://" in token
-            or (
-                token.count(".") >= 1
-                and any(c.isalpha() for c in token)
-                and "@" not in token
-            )
-        )
+        # Explicit URL indicators (http://, https://, www., or protocol markers)
+        if token.startswith(("http://", "https://", "www.")) or "://" in token:
+            return True
+
+        # Domain-like patterns (e.g., "example.com")
+        # But NOT abbreviations (e.g., "U.S.", "c.e.o.s")
+        # Heuristic: URLs have at least one period NOT followed by a single uppercase/lowercase letter
+        # This allows "example.com" but excludes "U.S." and "c.e.o.s"
+        if (
+            token.count(".") >= 1
+            and any(c.isalpha() for c in token)
+            and "@" not in token
+        ):
+            # Check if this looks like an abbreviation (single letters between periods)
+            # Pattern: letter(s).letter(s).letter(s) where segments are 1-3 chars
+            # Check if this looks like an abbreviation (single letters between periods)
```

Comment:
This comment seems redundant (seems like duplication of the line above)?

Reply:
Yeah - it got duplicated in error